fleetshard/pkg/fleetshardmetrics/metrics_test.go

```diff
@@ -0,0 +1,83 @@
+package fleetshardmetrics
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/expfmt"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMetricIncrements(t *testing.T) {
+	tt := []struct {
+		metricName    string
+		incrementFunc func()
+	}{
+		{
+			metricName:    "total_k8s_requests",
+			incrementFunc: IncrementK8sRequests,
+		},
+		{
+			metricName:    "total_k8s_request_errors",
+			incrementFunc: IncrementK8sRequestErrors,
+		},
+		{
+			metricName:    "total_fleet_manager_requests",
+			incrementFunc: IncrementFleetManagerRequests,
+		},
+		{
+			metricName:    "total_fleet_manager_request_errors",
+			incrementFunc: IncrementFleetManagerRequestErrors,
+		},
+	}
+	promParser := expfmt.TextParser{}
+
+	for _, tc := range tt {
+		t.Run(tc.metricName, func(t *testing.T) {
+			resetMetrics()
```

Comment:
Should this not be cleaned up after a test run instead of being required before?

```suggestion
			defer resetMetrics()
```